diagrams/sequence-diagrams/changing-a-form.md

```diff
@@ -9,25 +10,54 @@ sequenceDiagram
   link admin: GitHub repo @ https://github.com/alphagov/forms-admin   
   participant api as forms-api
   link api: GitHub repo @ https://github.com/alphagov/forms-api 
-  actor dev as GOV.UK Forms developer
 
   note right of user: Form has already been made live
 
-  user->>browser: Visit form page
-  browser->>admin: GET /forms/{form id}
-  admin->>api: GET /forms/{form id}
+  user->>browser: visit live form page
+  browser->>admin: GET /forms/{form id}/live
+  admin->>api: GET /api/v1/forms/{form id}
+  admin->>api: GET /api/v1/forms/{form id}/live
   browser-->>user: show live form page
   
-  note right of user: There is no option to edit the live form<br /><br />This is temporary behaviour whilst draft functionality is implemented.
-  
-  user->>dev: Request for form to be changed
-  dev->>api: Change form status from live to draft
-  dev-->>user: Inform user
-  
-  user->>browser: Visit form page
+  user->>browser: click "Create a draft to edit" button
   browser->>admin: GET /forms/{form id}
   admin->>api: GET /forms/{form id}
-  browser-->>user: Create a form
+  admin->>api: GET /forms/{form id}
+  admin->>api: GET /forms/{form id}/pages
+  browser-->>user: show draft form page with task list
   
-  note right of user: User now able to edit the form
+  note over user,api: User edits the draft form (See Creating form)<br>until they are ready to make their changes live
+
+  user->>browser: click "Make your changes live" task
+  browser->>admin: GET /forms/{form id}/make-live
+  admin->>api: GET /api/v1/forms/{form id}
+  admin->>api: GET /api/v1/forms/{form id}
+  browser-->>user: show "Make your changes live" page
+  alt form creator decides not to make form live
+    user->>browser: click "No" and submits page
+    browser->>admin: POST /forms/{form id}/make-live<br>payload: {forms_make_live_form"=><br>{"confirm_make_live"=>"not_made_live"},<br>"form_id"=>"{form id"}}
+    admin->>api: GET /api/v1/forms/{form id}
+    admin->>api: GET /api/v1/forms/{form id}
+    admin-->>browser: REDIRECT 302
+    browser->>admin: GET /forms/{form id}
+    admin->>api: GET /forms/{form id}
+    admin->>api: GET /forms/{form id}
+    admin->>api: GET /forms/{form id}/pages
+    browser-->>user: show draft form page with task list
+  else form creator decides to make form live
+    user->>browser: click "Yes" and submits page
+    browser->>admin: POST /forms/{form id}/make-live<br>payload: {forms_make_live_form"=><br>{"confirm_make_live"=>"made_live"},<br>"form_id"=>"{form id"}}
+    admin->>api: POST /api/v1/forms/{form id}/make-live<br>payload: {includes all form attributes and values}<br>Controller Action doesn't use any
```

Comment:
This endpoint eventually uses this model method to make a form live, as you can see we don't use any of the form attributes as forms-api sets the `live_at` using its own internal clock.

Reply:
https://github.com/alphagov/forms-api/blob/76f048e9d919d9a150109e5d9d543f24411ca364/app/models/form.rb#L9-L11